stroyprombeton/migrations/0019_catalog_data.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.20 on 2019-03-04 15:21
+from __future__ import unicode_literals
+
+from itertools import groupby
+from operator import attrgetter
+
+from django.db import migrations
+from django.db import transaction
+from django.db.migrations.operations.base import Operation
+
+FIELD_TO_TAG_GROUP = {
+    'specification': 'Серия',
+    'length': 'Длина',
+    'width': 'Ширина',
+    'height': 'Высота',
+    'weight': 'Масса',
+    'volume': 'Объём',
+    'diameter_out': 'Внешний диаметр',
+    'diameter_in': 'Внутренний диаметр',
+}
+
+TAG_FIELDS = FIELD_TO_TAG_GROUP.keys()
+
+OPTION_FIELDS = [
+    'date_price_updated', 'code', 'mark', 'is_new_price', 'price', 'in_stock',
+]
+
+
+class CreateOptionsOperation(Operation):
+    def state_forwards(self, app_label, state):
+        pass
+
+    @transaction.atomic()
+    def database_forwards(self, app_label, schema_editor, _, to_state):
+        product_model = to_state.apps.get_model(app_label, 'Product')
+        option_model = to_state.apps.get_model(app_label, 'Option')
+
+        grouped_products = groupby(
+            product_model.objects.all(),
```

Comment:
You have to sort products by `name` before, otherwise `groupby` won't work as expected

I wouldn't recommend use `groupby` at all because it is slow and implicit. `defaultdict(list)` looks more reliable